src/component/listitem.js

```diff
@@ -1,41 +1,36 @@
-import React, { useState } from "react";
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { toggleComplete, deleteTodo, editTodo } from '../redux/todoSlice';
 
 const ListItem = (props) => {
-
-    const checkedData = props.checked;
+    const dispatch = useDispatch();
     const data = props.item;
     const id = props.id;
+    const completed = props.checked;
 
     const [edit, setonEdit] = useState(false);
     const [editVal, setEditVal] = useState(data);
-    const [isChecked, setIsChecked] = useState(checkedData);
-
-    const onRemove = () => {
-        setTimeout(() => {
-            props.handleDelete();
-        }, 100)
+    const handleCompleteClick = () => {
+        dispatch(
+            toggleComplete({ id: id, completed: !completed }));
     }
-    const handelEditValue = e => setEditVal(e.target.value)
-    const onCheck = () => setIsChecked({ isChecked: true });
 
-    const handelEditOn = () => setonEdit(true)
+    const handleDeleteClick = () => {
+        dispatch(deleteTodo({ id: id }));
+    }
+    const handelEditValue = e => setEditVal(e.target.value);
+    const handelEditOn = () => setonEdit(true);
 
-    const handleCancel = () => setonEdit(false)
+    const handleCancel = () => setonEdit(false);
 
     const handleSave = () => {
         if (editVal === '') {
-            setEditVal(data)
+            dispatch(editTodo({ id: id, name: data }));
```

Comment:
If user does not enter any text then do not do anything

Reply:
if (editVal === '') {
            dispatch(editTodo({ id: id, name: data }));

this code means if user save empty in text box then default given data set